bundles/org.eclipse.swt/Eclipse SWT/win32/org/eclipse/swt/widgets/MenuItem.java

```diff
@@ -1151,17 +1176,12 @@ LRESULT wmMeasureChild (long wParam, long lParam) {
 
 	if ((parent.style & SWT.BAR) != 0) {
 		if (parent.needsMenuCallback()) {
-			/*
-			 * Weirdness in Windows. Setting `HBMMENU_CALLBACK` causes
-			 * item sizes to mean something else. It seems that it is
-			 * the size of left margin before the text. At the same time,
-			 * if menu item has a mnemonic, it's always drawn at a fixed
-			 * position. I have tested on Win7, Win8.1, Win10 and found
-			 * that value of 5 works well in matching text to mnemonic.
-			 * NOTE: autoScaleUpUsingNativeDPI() is used to avoid problems
-			 * with applications that disable automatic scaling.
-			 */
-			struct.itemWidth = DPIUtil.scaleUp(5, nativeZoom);
+			GC gc = new GC (display);
+			String textWithoutMnemonicCharacter = getText().replace("&", "");
+			Point points = gc.textExtent(textWithoutMnemonicCharacter);
+			gc.dispose ();
```

Comment:
Would be good to put this into a self-explaining method, such as `calculateRenderedTextSize()`.